### ISSUE TITLE
[BUG] Async Reprojection and Motion Smoothing unavailable using NVIDIA GPU

### ISSUE THREAD

**Issue by mimattr** [ValveSoftware/SteamVR-for-Linux#214]
#### Your system information
* Steam client version (build number or date): **Jul 18 2019. at 00:11:36**
* Distribution (e.g. Ubuntu): **Ubuntu 18.04.2 LTS (x64)**
* Graphics driver version (run nvidia-settings): **NVIDIA 430.26**
* [Gist for SteamVR System Information](https://gist.github.com/mimattr/9dd30e7e72dd953b8289eca9a66e31a9)
* Opted into Steam client beta?: **Yes**
* Opted into SteamVR beta?: **Yes**
* Have you checked for system updates?: **Yes**

#### Please describe your issue in as much detail as possible:
It appears `Asynchronous Reprojection` and `Motion Smoothing` are [unavailable when using NVIDIA GPUs](https://gist.github.com/mimattr/9dd30e7e72dd953b8289eca9a66e31a9#file-steamvr-txt-L3017) though no explanation is provided (tested using [Valve Index Headset](https://store.steampowered.com/app/1059530) and [GeForce RTX 2060](https://www.nvidia.com/en-gb/geforce/graphics-cards/rtx-2060)). Are these features currently in development or is this a driver/hardware limitation?

[Steam System Information](https://gist.github.com/mimattr/b0b7f46366e31e254460b9593ce87e51)

#### Steps for reproducing this issue:
1. Select the `Start SteamVR` icon within the Steam client (top-right location)
1. Select `Settings` from the STEAMVR application menu (top-left location)

**Comment by Vash63** [comment]
As of the last update I've heard from anyone affiliated with Valve, Async Reprojection is currently blocked on Nvidia's end. This was within 2 months, not sure I have the exact quote on hand though. I'm also fairly sure Motion Smoothing is dependent on Async Reprojection working (but also Motion Smoothing doesn't work on AMD either and is completely unimplemented in Linux).

**Comment by ChristophHaag** [comment]
As of https://steamcommunity.com/games/250820/announcements/detail/1612763361792731684 motion smoothing is supposedly working on AMD on Windows 10.

On Linux motion smoothing indeed does not exist yet.

**Comment by Pivert** [comment]
Hi, do we have any update on this ? This feature is really required for CPU intensive games. This is required for X-Plane VR on Linux for instance.

**Comment by leinardi** [comment]
Half-Life: Alyx is officially out for Linux but still no async reprojection for Nvidia users...

**Comment by mimattr** [comment]
> This Motion Smoothing and Async Reprojection are really required for CPU intensive games.

> Half-Life: Alyx is officially out for Linux but still no async reprojection for Nvidia users.

@Pivert @leinardi Valve recently confirmed it's an [NVIDIA driver limitation](https://twitter.com/dan_ginsburg/status/1261403868279140353), hopefully this long-standing issue will be resolved shortly following the official release of [Half-Life: Alyx](https://store.steampowered.com/app/546560) on Linux.

**Comment by SlickMcRunFast** [comment]
RANT WARNING

As a clear indicator of nvidias dedication to Linux, check out their installer (ncurses...) used to scare new users away for the past two decades. Leads me to believe nvidia will provide async projection when Steam VR for Linux learns to turn off my light houses and switch audio to my PC speaker.

My next card will be AMD.

**Comment by mnn** [comment]
@SlickMcRunFast Believe me, I know the pain - https://www.reddit.com/r/linux_gaming/comments/gkxfwd/is_vr_on_linux_halfbaked_or_is_something_wrong/.

> My next card will be AMD.

If AMD comes with comparable hi-end card to NVidia, yeah, I am most likely switching teams too. (When I was picking a GPU, AMD's best was 5700xt.) Unless NVidia releases their drivers as open-source, but, well, I am not holding my breath.

**Comment by KindDragon** [comment]
Maybe we should report this to NVidia https://www.nvidia.com/en-us/geforce/forums/game-ready-drivers/13/ ?

**Comment by cosmicchipsocket** [comment]
Correct me if I'm wrong but isn't the "Game-Ready Drivers" moniker more for the Windows specific drivers? Are you sure the issue would get visibility there?

This issue could definitely use a poke though. I want to upgrade to a raytracing card at some point and VR driver support will be influencing whether I go with nVidia or AMD this time around.

**Comment by KindDragon** [comment]
@cosmicchipsocket You right, but NVidia suggest that bug reports should go there or directly to Nvidia Support https://www.nvidia.com/en-us/support/consumer/

**Comment by guysoft** [comment]
Yes I am using Nvidia, I have a Asus Zephyrus g14 with an NVIDIA RTX 2060 Max-Q.
It also has an AMD on the integrated CPU. But its not strong enough to push VR.
Since this is a laptop I can't really switch cards neither.

**Comment by TiagoTiago** [comment]
Would there be a way to implement similar functionality without the help of Nvidia?

**Comment by KindDragon** [comment]
This is a forum for Linux driver https://forums.developer.nvidia.com/c/gpu-unix-graphics/linux/148

**Comment by DASPRiD** [comment]
> Would there be a way to implement similar functionality without the help of Nvidia?

No, this does require a certain Vulkan extension.

**Comment by KindDragon** [comment]
https://www.gamingonlinux.com/2021/06/nvidia-driver-470-for-linux-to-include-support-for-async-reprojection

**Comment by Gusted** [comment]
FYI:

The 470 _beta_ driver has been released: https://www.nvidia.com/download/driverResults.aspx/176525/en-us
And for some reason SteamVR _beta_ had already added support for compatible 470 drivers a week ago ago: https://store.steampowered.com/news/app/250820/view/3007823106776506362

**Comment by bndlfm** [comment]
Just to confirm async reprojection has been enabled on NVIDIA cards with SteamVR beta 1.18.4 and NVIDIA 470?

I'm on SteamVR beta 1.18.4 and NVIDIA 470.42.01-1 and I'm unable to enable motion smoothing w/ a gtx 1660. Should I open a new issue or post all my info here?

**Comment by mimattr** [comment]
> Should I open a new issue or post all my info here?

@bndlfm Please post your "Motion Smoothing" issue(s) here, thanks.

**Comment by Vash63** [comment]
Can someone edit the bug title to remove "Async Reprojection" since that's fixed now? Also Motion Smoothing isn't available on AMD or Intel GPUs, so this could really just be simplified to "Motion Smoothing unavailable on Linux".

**Comment by mimattr** [comment]
> Can someone edit the bug title to remove "Async Reprojection" since that's fixed now?

@Vash63 Will update the title once the **NVIDIA 470** graphics driver is officially available (and tested) from the standard [Ubuntu repository/PPA](https://launchpad.net/~graphics-drivers/+archive/ubuntu/ppa)

**Comment by culturedrock** [comment]
I'm not certain about the split between Async Reprojection and Motion Smoothing here - is the latter supposed to be already enabled while using 470 drivers? I'm on SteamVR beta 1.19.7, NVidia 470.63.01, kernel 5.10.61-1-lts (Arch Linux x64), still getting "Motion Smoothing: Not Supported".

**Comment by ZarathustraDK** [comment]
@culturedrock Async Reprojection and motion smoothing are 2 different things. I'm sure there's a more technical description of it somewhere, but basically Async Reprojection lets you pivot your head around and it'll feel smooth even if the scene itself is not running smoothly. For instance, in Blade and Sorcery a lot of enemies onscreen will still reduce the framerate they're moving at, but YOU wont feel you're moving your head at a reduced framerate/lagging behind, which helps a lot with VR-sickness if working as intended.
Motion smoothing, which has always been off on linux, tries to interpolate between missing frames in the scene when you move laterally, and it can become quite jarring and feel like your vision is being messed with if egregious enough. It's not a loss that it stays off in my book.

**Comment by Adrenocrom** [comment]
Motion smoothing is still the missing feature until today!
We need it for hardware intensive applications, there are open source solutions for video smoothing can they be integrated in the pipeline?

**Comment by XDM-Inc** [comment]
still not working???

**Comment by sheepdestroyer** [comment]
really?
